plugins/inputs/vsphere/endpoint.go

```diff
@@ -588,6 +591,17 @@ func (e *Endpoint) Collect(ctx context.Context, acc telegraf.Accumulator) error
 }
 
 func (e *Endpoint) chunker(ctx context.Context, f PushFunc, res *resourceKind, now time.Time, latest time.Time) {
+	maxMetrics := e.Parent.MaxQueryMetrics
+	if maxMetrics < 1 {
+		maxMetrics = 1
+	}
+
+	// Workaround for vCenter weirdness. Cluster metrics seem to count multiple times
+	// when checking query size, so keep it at a low value.
+	// Revisit this when we better understand the reason why vCenter counts it this way!
+	if res.name == "cluster" && maxMetrics > 10 {
+		maxMetrics = 10
```

Comment:
I'm not changing the setting, just setting a local variable for this turn of the loop that goes through resource types. This is a bit of an ugly hack, since clusters seem to need special treatment for some reason. We're investigating this, but this hack solves the immediate problem.